mne/decoding/gsoc.py

```diff
@@ -0,0 +1,195 @@
+# Authors: Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from .mixin import TransformerMixin
+from ..epochs import _BaseEpochs
+
+
+class EpochsTransformerMixin(TransformerMixin):
+    """Mixin class for reshaping data to Epoch's standard shape
+
+    This class is meant to be inherited by transformers meant to be
+    used in scikit-learn pipeline. Useful utility for making data
+    compatible with mne internal functions.
+
+    Parameters
+    ----------
+    n_chan : int | None
+
+        The number of channels.
+    """
+
+    def __init__(self, n_chan=None):
+        self.n_chan = n_chan
+        self._check_init()
+
+    def fit(self, X, y=None):
+        """No use here. Added for scikit-learn compatibility.
+
+        Parameters
+        ----------
+        X : numpy array of dimensions [2,3,4]
+            The data to be reshaped.
+        y : None
+            Used for scikit-learn compatibilty
+
+        Returns
+        -------
+        self : Instance of EpochsTransformerMixin
+            Return the same object.
+        """
+        return self
+
+    def fit_transform(self, X, y=None):
```

Comment:
Same here, `fit_transform` may be directly inheritable from TransformerMixin.

Else, change the description "no use here" isn't quite correct. `fit` `fit_transform` will be used by the transformers.